client/dashboard/sites/deployments/dataviews/fields.tsx

```diff
@@ -0,0 +1,160 @@
+import { Badge } from '@automattic/ui';
+import {
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useMemo } from 'react';
+import TimeSince from '../../../components/time-since';
+import {
+	DeploymentStatusBadge,
+	DeploymentStatusValue,
+} from '../components/deployment-status-badge';
+import type { DeploymentRunWithDeploymentInfo } from '../hooks/use-code-deployment-runs-query';
+import type { Field } from '@wordpress/dataviews';
+
+export function useDeploymentFields(): Field< DeploymentRunWithDeploymentInfo >[] {
+	return useMemo(
+		() => [
+			{
+				id: 'repository_name',
+				label: __( 'Repository' ),
+				enableHiding: false,
+				enableGlobalSearch: true,
+				getValue: ( { item } ) => item.repository_name,
+				render: ( { item } ) => {
+					const [ , repo ] = item.repository_name.split( '/' );
+					return <Text>{ repo }</Text>;
+				},
+			},
+			{
+				id: 'commit',
+				label: __( 'Commit' ),
+				enableGlobalSearch: true,
+				enableSorting: false,
+				getValue: ( { item } ) => {
+					return item.metadata?.commit_message || '';
+				},
+				render: ( { item } ) => {
+					if ( ! item.metadata ) {
+						return <Text variant="muted">{ __( 'No commit info' ) }</Text>;
+					}
+
+					const { commit_message, commit_sha, author } = item.metadata;
+					const [ installation, repo ] = item.repository_name.split( '/' );
+					const shortSha = commit_sha?.substring( 0, 7 ) || '';
+					const color = '#3b3b3b';
+
+					return (
+						<VStack spacing={ 1 }>
+							<Text title={ commit_message }>{ commit_message }</Text>
+							<HStack spacing={ 3 } alignment="left" style={ { width: 'auto' } }>
+								<a
+									href={ `https://github.com/${ installation }/${ repo }/commit/${ commit_sha }` }
+									target="_blank"
+									rel="noopener noreferrer"
+								>
+									<Text
+										as="code"
+										size="small"
+										style={ { color: 'var(--wp-admin-theme-color-darker-20)' } }
+									>
+										{ shortSha }
+									</Text>
+								</a>
+								<HStack spacing={ 1 } alignment="left" style={ { width: 'auto' } }>
+									<svg
```

Comment:
Could we move it to a separated file 😅 We can set the fill to `currentColor` and control the color from the parent.